### ISSUE TITLE
diagnostic::do_not_recommend can emit an error if it encounters a keyword

### ISSUE THREAD

**Issue by mejrs** [rust-lang/rust#151996]
I tried this code:

```rust
#![crate_type = "lib"]

pub trait X{}
pub struct Y;

#[diagnostic::do_not_recommend(if)]
impl X for Y {}
```

I expected to see this happen: *a warning*

Instead, this happened: *an error*

### Meta

```
current nightly
```

this attribute is never allowed to error, this is a corner case I missed in https://github.com/rust-lang/rust/pull/151516

cc @JonathanBrouwer @jdonszelmann 

@rustbot claim
